### PR TITLE
build(Makefile): more docker env

### PR DESCRIPTION
**Pull request by ovalenti**
### Description

This is a small change to the main Makefile that forwards `SHORTCOMMIT` inside the context of dockerized build. This simplifies building in git worktrees, where git is not available from within the container.

### Testing

I expect those changes to be no-op in the usual build environment.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ ROX_IMAGE_FLAVOR ?= $(shell \
 	  echo "development_build"; \
 	fi)
 
-DEFAULT_IMAGE_REGISTRY := quay.io/stackrox-io
```
Comment (msugakov) [inline Makefile:51]: You can still override `DEFAULT_IMAGE_REGISTRY` when it's set the old way:
```bash
make DEFAULT_IMAGE_REGISTRY=quay.io/blah/fo-bar my-target
# or
make my-target DEFAULT_IMAGE_REGISTRY=quay.io/blah/fo-bar
```
I.e. use make variable, not env variable.

What is your use-case, do you do it to be able to push containers in your private repos?

**Review by msugakov — COMMENTED** [review]
Not a blocking feedback, more wondering about the use-case.

File: Makefile (hunk 1 above)
Comment (ovalenti) [reply Makefile:51]: Yes, this is my use-case. TIL that these do not have the same behavior :
- `make DEFAULT_IMAGE_REGISTRY=quay.io/blah/fo-bar my-target`
- `DEFAULT_IMAGE_REGISTRY=quay.io/blah/fo-bar make my-target`

While they are equivalent when using `?=`

Comment (ovalenti) [reply Makefile]: Thank you for the review @msugakov .
I reverted this part of the PR, to keep only the docker envvar.

**Review by msugakov — APPROVED** [review]

**Pull request merged**